galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,214 @@
+package dev.galasa.framework.api.webui;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import dev.galasa.framework.spi.DssAdd;
+import dev.galasa.framework.spi.DssSwap;
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+import dev.galasa.framework.spi.FrameworkPropertyFileException;
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResultArchiveStoreDirectoryService;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+
+public class WorklistUtils {
+	
+	private IFramework framework;
+	
+	public WorklistUtils(IFramework framework) {
+		this.framework = framework;
+	}
+	
+	
+	protected List<String> getRunIdsFromDss() throws DynamicStatusStoreException, ServletException {
+	
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String worklistProperty = dss.get("user.username.worklist");
```

Comment:
Get the "username" from the SecurityContext please

Reply:
if it is null,  or the principal is null or the username of the principal,  set to "unknown"